src/numeric/impl_numeric.rs

```diff
@@ -133,17 +161,23 @@ impl<A, S, D> ArrayBase<S, D>
     /// let a = arr2(&[[1., 2.],
     ///                [3., 4.]]);
     /// assert!(
-    ///     a.mean_axis(Axis(0)) == aview1(&[2.0, 3.0]) &&
-    ///     a.mean_axis(Axis(1)) == aview1(&[1.5, 3.5])
+    ///     a.mean_axis(Axis(0)).unwrap() == aview1(&[2.0, 3.0]) &&
+    ///     a.mean_axis(Axis(1)).unwrap() == aview1(&[1.5, 3.5])
     /// );
     /// ```
-    pub fn mean_axis(&self, axis: Axis) -> Array<A, D::Smaller>
+    pub fn mean_axis(&self, axis: Axis) -> Option<Array<A, D::Smaller>>
         where A: Clone + Zero + FromPrimitive + Add<Output=A> + Div<Output=A>,
               D: RemoveAxis,
     {
-        let n = A::from_usize(self.len_of(axis)).expect("Converting axis length to `A` must not fail.");
-        let sum = self.sum_axis(axis);
-        sum / &aview0(&n)
+        let axis_length = self.len_of(axis.clone());
```

Comment:
`Axis` is `Copy`; `.clone()` isn't necessary.

```suggestion
        let axis_length = self.len_of(axis);
```